plugins/modules/zabbix_autoregistration.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: zabbix_autoregistration
+
+short_description: Update Zabbix autoregistration
+
+
+description:
+    - This module allows you to modify Zabbix autoregistration.
+
+author:
+    - ONODERA Masaru(@masa-orca)
+
+requirements:
+    - "zabbix-api >= 0.5.4"
+
+options:
+    tls_accept:
+        description:
+            - Type of allowed incoming connections for autoregistration.
+            - Choose from C(unsecure), C(tls_with_psk) or both.
+        type: list
+        elements: str
+        required: true
+    tls_psk_identity:
+        description:
+            - TLS connection use this PSK identity string.
+            - The PSK identity string will be transmitted unencrypted over the network. Therefore, you should not put sensitive information.
+            - It is required if you set C(tls_with_psk) to I(tls_accept) and current tls_accept is C(unsecure).
+        type: str
+    tls_psk:
+        description:
+            - TLS connection use this PSK value.
+            - It is required if you set C(tls_with_psk) to I(tls_accept) and current tls_accept is C(unsecure).
```

Comment:
```suggestion
            - TLS connection uses this PSK value.
            - This setting requires I(tls_accept=tls_with_psk) if current value of I(tls_accept) is C(unsecure)
```

Does the 2nd line mean that `tls_accept=tls_with_psk` is only required when you updating from `tls_accept=unsecure`, but can be omitted when re-running multiple times without module doing any changes itself?

Reply:
I tried it. However, I could not come up with best way to get last task's arguments.
In addition, I think these values must be not logged.